posthog/api/dashboard.py

```diff
@@ -130,6 +131,12 @@ def retrieve(self, request: Request, *args: Any, **kwargs: Any) -> response.Resp
         serializer = DashboardSerializer(dashboard, context={"view": self, "request": request})
         return response.Response(serializer.data)
 
+    def get_parents_query_dict(self) -> Dict[str, Any]:  # to be moved to a separate Legacy*ViewSet Class
+
+        if not self.request.user.is_authenticated or "share_token" in self.request.GET:
+            return {}
```

Comment:
:thinking: So unauthenticated people get access to all of the data?

Not sure the behavior here is exactly identical than before.

Reply:
They don't, this this is basically an override so that more "manual" filtering can be done in `DashboardsViewSet`'s `get_queryset` instead of having `NestedViewSetMixin` do that as usual (as it wouldn't work in this case).